homeassistant/components/homeassistant_sky_connect/__init__.py

```diff
@@ -64,18 +63,9 @@ async def _multi_pan_addon_info(
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up a Home Assistant Sky Connect config entry."""
-    matcher = usb.USBCallbackMatcher(
-        domain=DOMAIN,
-        vid=entry.data["vid"].upper(),
-        pid=entry.data["pid"].upper(),
-        serial_number=entry.data["serial_number"].lower(),
-        manufacturer=entry.data["manufacturer"].lower(),
-        description=entry.data["description"].lower(),
-    )
-
-    if not usb.async_is_plugged_in(hass, matcher):
-        # The USB dongle is not plugged in
-        raise ConfigEntryNotReady
+    if not await async_is_plugged_in(hass, entry):
+        # The USB dongle is not plugged in, don't setup ZHA
+        return True
```

Comment:
It's strange that we notify listeners of a successful entry load when it didn't actually succeed. Should this be `False`?

Reply:
I guess they would have to restart if they plug it in after startup?

Reply:
If we unload the config entry if the dongle is not plugged in when Home Assistant starts, who is then loading the config entry if the dongle is plugged in again?

Reply:
USB discovery?

Reply:
That's essentially the same as https://github.com/home-assistant/core/pull/84975, although that PR *disabled* the config entry during setup and *enabled* it during discovery.
It's not a reliable solution, because unloading the config entry will race against it being loaded from the config flow.

@bdraco suggested adding a new optional function `async_is_present` to components, which will avoid unloading or disabling the config entry from a task:

```diff
diff --git a/homeassistant/components/homeassistant_sky_connect/__init__.py b/homeassistant/components/homeassistant_sky_connect/__init__.py
index 4315862e52..15b20b0af2 100644
--- a/homeassistant/components/homeassistant_sky_connect/__init__.py
+++ b/homeassistant/components/homeassistant_sky_connect/__init__.py
@@ -112,3 +112,8 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload a config entry."""
     return True
+
+
+async def async_is_present(hass: HomeAssistant) -> bool:
+    """Return True if the Sky Connect is present."""
+    return True
\ No newline at end of file
diff --git a/homeassistant/config_entries.py b/homeassistant/config_entries.py
index 2907b6a966..1208dec45d 100644
--- a/homeassistant/config_entries.py
+++ b/homeassistant/config_entries.py
@@ -379,6 +379,11 @@ class ConfigEntry:
 
         error_reason = None
 
+        if hasattr(component,"async_is_present"):
+            present = await component.async_is_present(hass, self)
+            if not present:
+                --- disable here ---
+
         try:
             result = await component.async_setup_entry(hass, self)
```